### PR TITLE
[RIVER-1701] Remove support for legacy (unmigrated) stream schema

### PR DESCRIPTION
**Pull request by clemire**
#1701

### CODE REVIEW

File: core/node/storage/pg_stream_store.go [hunk 1]
```diff
@@ -376,7 +376,7 @@ func (s *PostgresStreamStore) createStreamArchiveStorageTx(
 			`INSERT INTO es (stream_id, latest_snapshot_miniblock, migrated) VALUES ($1, -1, false);
 			CREATE TABLE {{miniblocks}} PARTITION OF miniblocks FOR VALUES IN ($1);`,
 			streamId,
-			false,
+			s.config.MigrateStreamCreation,
```
Comment (clemire) [inline core/node/storage/pg_stream_store.go:379]: Bugfix: the archiver was always using the legacy schema.

File: core/node/testutils/dbtestutils/db.go [hunk 2]
```diff
@@ -76,14 +76,15 @@ func ConfigureDB(ctx context.Context) (*config.DatabaseConfig, string, func(), e
...
-			NumPartitions: 4,
+			Host:                  "localhost",
+			Port:                  5433,
+			User:                  "postgres",
+			Password:              "postgres",
+			Database:              "river",
+			Extra:                 "?sslmode=disable&pool_max_conns=1000",
+			StartupDelay:          2 * time.Millisecond,
+			NumPartitions:         4,
+			MigrateStreamCreation: true,
```
Comment (sergekh2) [inline core/node/testutils/dbtestutils/db.go:87]: after this change it seems this setting is no longer required?

**Review by sergekh2 — APPROVED** [review]

**Review by bas-vk — APPROVED** [review]

File: core/node/testutils/dbtestutils/db.go (hunk 2 above)
Comment (clemire) [reply core/node/testutils/dbtestutils/db.go:87]: Yes, I was going to do in a follow-up PR but went ahead and put in here so the code would not be in a confusing state.

**Comment by clemire** [comment]
@sergekh2 I updated this PR according to your comment to remove the migration setting, and i also updated the migration tool, but I did not test the migration tool since we don't really need it right now.  I saw some flakes locally but after making changes they disappeared - if 5 consecutive CI tests pass I will consider this solid.

**Comment by clemire** [comment]
> @sergekh2 I updated this PR according to your comment to remove the migration setting, and i also updated the migration tool, but I did not test the migration tool since we don't really need it right now. I saw some flakes locally but after making changes they disappeared - if 5 consecutive CI tests pass I will consider this solid.

Ok, all 5 runs after a lint fix are totally green.

**Comment by clemire** [comment]
Note: I can drop the "migrated" column in a follow-up PR, I did not do that here.

**Review by sergekh2 — APPROVED** [review]

**Pull request merged**